UnitsNet/CustomCode/QuantityInfo/QuantityInfoExtensions.cs

```diff
@@ -9,6 +9,22 @@ namespace UnitsNet;
 /// </summary>
 internal static class QuantityInfoExtensions
 {
+    /// <summary>
+    ///     Get a list of quantities having the given base dimensions.
+    /// </summary>
+    /// <param name="quantityInfos">The type of quantity mapping information.</param>
+    /// <param name="baseDimensions">The base dimensions to match.</param>
+    public static IEnumerable<QuantityInfo> GetQuantitiesWithBaseDimensions(this IEnumerable<QuantityInfo> quantityInfos,
+        BaseDimensions baseDimensions)
+    {
+        if (baseDimensions is null)
+        {
+            throw new ArgumentNullException(nameof(baseDimensions));
+        }
+
+        return quantityInfos.Where(info => info.BaseDimensions.Equals(baseDimensions));
+    }
```

Comment:
This extensions doesn't have to be public: it's currently only used from the `Quantity` class.